modules/calc/src/main/java/com/opengamma/strata/calc/config/ReportingCurrency.java

```diff
@@ -23,63 +24,85 @@
 import org.joda.beans.impl.direct.DirectMetaProperty;
 import org.joda.beans.impl.direct.DirectMetaPropertyMap;
 
-import com.opengamma.strata.basics.CalculationTarget;
 import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.calc.runner.function.CalculationFunction;
 
 /**
- * Reporting rules that always returns the same currency.
+ * The reporting currency.
  * <p>
- * These rules always return the same currency from {@link #reportingCurrency}.
+ * This is used to specify the currency that the result should be reporting in.
+ * The currency specified may be explicit, using {@link #of(Currency)}, or implicit
+ * using {@link #NATURAL}.
  */
 @BeanDefinition(builderScope = "private")
-final class FixedReportingRules implements ReportingRules, ImmutableBean {
+public final class ReportingCurrency
+    implements ImmutableBean, Serializable {
 
   /**
-   * The reporting currency that is used for every target.
+   * An instance requesting the "natural" currency of the target.
+   * <p>
+   * When converting calculation results, conversion will occur to the "natural" currency of the target.
+   * The "natural" currency of a target is obtained from {@link CalculationFunction}.
+   */
+  public static final ReportingCurrency NATURAL = new ReportingCurrency(ReportingCurrencyType.NATURAL, null);
+
+  /**
+   * The type of reporting currency.
    */
   @PropertyDefinition(validate = "notNull")
+  private final ReportingCurrencyType type;
```

Comment:
This feels redundant. The caller already has two ways to find out whether a specific currency is avilable:

    reportingCurrency == ReportingCurrency.NATURAL;

and

    resportingCurrency.getCurrency().isPresent();

I don't think we need this too.